Tests/Shared/StressTestCase.swift

```diff
@@ -17,23 +17,22 @@ class StressTestCase: XCTestCase {
     }
     
     override class func tearDown() {
-        // After running test suite cleanup `NSTemporaryDirectory()` and `Documents` directory
-        let fileManager = FileManager.default
-        let tempDirectoryURL = URL(fileURLWithPath: NSTemporaryDirectory(), isDirectory: true)
-        let documentsURL = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first
-        let tempFileURLs = (try? fileManager.contentsOfDirectory(
-            at: tempDirectoryURL,
-            includingPropertiesForKeys: nil,
-            options: .skipsHiddenFiles
-        )) ?? []
-        let documentsFileURLs = documentsURL.flatMap { try? FileManager.default.contentsOfDirectory(
-            at: $0,
-            includingPropertiesForKeys: nil,
-            options: .skipsHiddenFiles
-        ) } ?? []
-        
-        for fileURL in tempFileURLs + documentsFileURLs {
-            try? fileManager.removeItem(at: fileURL)
+        do {
+            // After running test suite cleanup `NSTemporaryDirectory()` and `Documents` directory
+            let fileManager = FileManager.default
+            let tempDirectoryURL = URL(fileURLWithPath: NSTemporaryDirectory(), isDirectory: true)
+            
+            let tempFileURLs = try fileManager.contentsOfDirectory(
+                at: tempDirectoryURL,
+                includingPropertiesForKeys: nil,
+                options: .skipsHiddenFiles
+            )
+            
+            try tempFileURLs.forEach {
+                try fileManager.removeItem(at: $0)
+            }
+        } catch {
+            XCTFail("Failed to clean up after tests: \(error)")
```

Comment:
I don't like relying on `StressTestCase` that it will cleanup temp directory, e.g. originally tests in #923 were failing because of that and `ChannelDTO_Tests` became `StressTestCase` because of my incorrect commit as I just used it for debugging, but never intended to push it on purpose. 

I'd prefer deleting DB file in `DatabaseContainer_Mock.deinit`, to generally prevent orphaned DB files. I know you're aware of general tests performance, so if you're aware of this limitation, it's good with me.

Reply:
@olejnjak Makes sense 👍 I added the general cleanup to `TestsEnvironmentSetup` to make sure we always start with a clean temp folder. Then I added a flag to `DatabaseContainerMock` that makes it delete its own DB files once it's deinited. I restricted this functionality only to the convenience init since there are some tests that rely on the fact the file stays there. The assumption is: if you use the convenience init, you don't really on the DB details in the tests.